### PR TITLE
[EP-594] Added iDEAL Polling and response handling

### PR DESCRIPTION
**Pull request by mpetrenco**
- Added `iDEAL Transaction Details` section to the Objective-C sample app;
- Added checks for ApplePay (no configuration, not supported);
- Added checks for iDEAL (currency not EUR);
- Added `sale` and `statusrequest` calls;
- Added the `TransactionStatusView`;
- Fixed `Currency Switcher` not working on the Objective-C sample app;
- Updated `JPReference` length logic to match `develop` branch

### CODE REVIEW

File: Source/IDEALViewController/TransactionStatusView/JPTransactionStatusView.m [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        case JPTransactionStatusTimeout:
+            self.titleLabel.text = @"transaction_timeout".localized;
+            break;
+    }
+}
+
+#pragma mark - Layout Setup
+
+- (void)setupViews {
+    self.backgroundColor = [UIColor colorWithWhite:0.0 alpha:0.3];
```
Comment (gcojocaru) [inline Source/IDEALViewController/TransactionStatusView/JPTransactionStatusView.m]: Can we extract the color as a constant?

File: Source/IDEALViewController/TransactionStatusView/JPTransactionStatusView.m [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        _titleLabel.textColor = UIColor.jpBlackColor;
+    }
+    return _titleLabel;
+}
+
+- (UIView *)containerView {
+    if (!_containerView) {
+        _containerView = [UIView new];
+        _containerView.translatesAutoresizingMaskIntoConstraints = NO;
+        _containerView.backgroundColor = [UIColor colorFromHex:0xE4E4E4];
```
Comment (gcojocaru) [inline Source/IDEALViewController/TransactionStatusView/JPTransactionStatusView.m]: Can we extract the color as a constant?

File: Source/Modules/PaymentMethods/View/Subviews/Cells/EmptyCardListCell/JPPaymentMethodsEmptyCardListCell.m [hunk 3]
```diff
@@ -84,11 +85,16 @@ - (void)setupViews {
...
 
 - (void)setupConstraints {
-    NSLayoutConstraint *heightConstraint = [self.addCardButton.heightAnchor constraintEqualToConstant:36.0f];
-    heightConstraint.priority = 999;
-    heightConstraint.active = YES;
-
-    [self.stackView pinToView:self withPadding:0.0f];
+    NSArray *constraints = @[
+        [self.stackView.topAnchor constraintEqualToAnchor:self.topAnchor
+                                                 constant:60.0],
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/EmptyCardListCell/JPPaymentMethodsEmptyCardListCell.m]: Can we extract this to a constant?

File: Source/Modules/PaymentMethods/View/Subviews/Cells/EmptyCardListCell/JPPaymentMethodsEmptyCardListCell.m [hunk 4]
```diff
@@ -84,11 +85,16 @@ - (void)setupViews {
...
+        [self.stackView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+        [self.stackView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
+        [self.stackView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+        [self.addCardButton.heightAnchor constraintEqualToConstant:36.0f],
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/EmptyCardListCell/JPPaymentMethodsEmptyCardListCell.m]: Can we extract this to a constant?

File: Source/Modules/PaymentMethods/View/Subviews/Cells/EmptyCardListCell/JPPaymentMethodsEmptyCardListCell.m [hunk 5]
```diff
@@ -84,11 +85,16 @@ - (void)setupViews {
...
+    ];
+
+    [NSLayoutConstraint activateConstraints:constraints withPriority:999];
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/EmptyCardListCell/JPPaymentMethodsEmptyCardListCell.m]: Can we extract this to a constant?

File: Source/Modules/PaymentMethods/Builder/JPPaymentMethodsBuilder.m [hunk 6]
```diff
@@ -44,6 +45,24 @@ + (JPPaymentMethodsViewController *)buildModuleWithMode:(TransactionMode)mode
                                   transitioningDelegate:(JPSliderTransitioningDelegate *)transitioningDelegate
                                       completionHandler:(JudoCompletionBlock)completion {
 
+    for (JPPaymentMethod *paymentMethod in configuration.paymentMethods) {
+        BOOL isIDEALPresent = (paymentMethod.type == JPPaymentMethodTypeIDeal);
+        BOOL isApplePayPresent = (paymentMethod.type == JPPaymentMethodTypeApplePay);
+        BOOL isCurrencyEUR = [configuration.amount.currency isEqualToString:@"EUR"];
```
Comment (stpopa) [inline Source/Modules/PaymentMethods/Builder/JPPaymentMethodsBuilder.m]: let's extract `@"EUR"` to a constant

File: Source/Modules/IDEALViewController/JPIDEALViewController.m [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+- (void)handleResponse:(JPResponse *)response {
+    JPOrderDetails *orderDetails = response.items.firstObject.orderDetails;
+    if (orderDetails && [orderDetails.orderFailureReason isEqualToString:@"USER_ABORT"]) {
+        [self dismissViewControllerAnimated:YES completion:nil];
+        self.completionBlock(response, NSError.judoUserDidCancelError);
+        return;
+    }
+
+    NSString *amountString = [NSString stringWithFormat:@"%.2f", orderDetails.amount];
+    response.items.firstObject.amount = [JPAmount amount:amountString currency:@"EUR"];
```
Comment (stpopa) [inline Source/Modules/IDEALViewController/JPIDEALViewController.m]: please extract `@"EUR"` and `@"USER_ABORT"` to global constants
you already have defined `static NSString * const kEuroCurrency = @"EUR";` just make it visible and reuse it here and in `JPPaymentMethodsInteractor.m`

**Pull request merged**